cpp/src/arrow/python/pyarrow.cc

```diff
@@ -57,9 +57,6 @@ int import_pyarrow() {
     } else {                                                                            \
       return UnwrapError(obj, #TYPE_NAME);                                              \
     }                                                                                   \
-  }                                                                                     \
-  Status unwrap_##FUNC_SUFFIX(PyObject* obj, std::shared_ptr<TYPE_NAME>* out) {         \
-    return unwrap_##FUNC_SUFFIX(obj).Value(out);                                        \
```

Comment:
This removal broke the turbodbc integration tests. Apparently turbodbc was still using the version with `*out`, eg at https://github.com/blue-yonder/turbodbc/blob/be96fef1450de9cd77a506fdce141dda563fd8b3/cpp/turbodbc_arrow/Library/src/set_arrow_parameters.cpp#L427

Reply:
Opened ARROW-13594 to track the build failure on our side